src/huggingface_hub/hf_api.py

```diff
@@ -1559,16 +1558,7 @@ def delete_repo(
             headers={"authorization": f"Bearer {token}"},
             json=json,
         )
-        try:
-            _raise_for_status(r)
-        except requests.exceptions.RequestException as e:
-            try:
-                message = e.response.json()["error"]
-            except JSONDecodeError:
-                message = e.response.text
-            except AttributeError:
-                message = e.args[0]
-            raise type(e)(message) from e
+        _raise_for_status(r)
```

Comment:
This was causing issues. The existing fix would raise an error such as 

```
huggingface_hub.utils._errors.RepositoryNotFoundError: 404 Client Error: Repository Not Found for url: https://huggingface.co/api/models/efwfwefw. If the repo is private, make sure you are authenticated. (Request ID: ktudRx2-3fp48WpTK6013)
```
but the `e.response.text` is still `Repository not found`.

Reply:
Still to be figured would be this though

```py
try:
   model_info("efwfwefw")
except RepositoryNotFoundError as e:
   print(e.response.text)
```

```
{"error":"Repository not found"}
```

but 
```python
model_info("efwfwefw")
```

correctly gives

```py
huggingface_hub.utils._errors.RepositoryNotFoundError: 404 Client Error: Repository Not Found for url: https://huggingface.co/api/models/efwfwefw. If the repo is private, make sure you are authenticated. (Request ID: WJtGL2IulzooY9JLwHEBn)
```